src/Symfony/Component/Translation/Bridge/Phrase/PhraseProvider.php

```diff
@@ -0,0 +1,276 @@
+                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+
+                // using weak etags, responses for requests with fallback locale enabled can not be reliably cached...
+                if (false === $this->readConfig->isFallbackLocaleEnabled()) {
+                    $headers = $response->getHeaders(false);
+                    $item->set(new PhraseCachedResponse($headers['etag'][0], $headers['last-modified'][0], $content));
+                    $this->cache->save($item);
+                }
+            }
+        }
+
+        $event = new PhraseReadEvent($translatorBag);
+        $this->dispatcher->dispatch($event);
+
+        return $event->getBag();
+    }
+
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        \assert($translatorBag instanceof TranslatorBag);
+
+        $event = new PhraseWriteEvent($translatorBag);
+        $this->dispatcher->dispatch($event);
+
+        $catalogues = $event->getBag()->getCatalogues();
+
+        foreach ($catalogues as $catalogue) {
+            foreach ($catalogue->getDomains() as $domain) {
+                if (0 === \count($catalogue->all($domain))) {
+                    continue;
+                }
+
+                $phraseLocale = $this->getLocale($catalogue->getLocale());
+
+                $content = $this->xliffFileDumper->formatCatalogue($catalogue, $domain, ['default_locale' => $this->defaultLocale]);
+                $filename = sprintf('%d-%s-%s.xlf', date('YmdHis'), $domain, $catalogue->getLocale());
+
+                $fields = array_merge($this->writeConfig->setTag($domain)->setLocale($phraseLocale)->getOptions(), ['file' => new DataPart($content, $filename, 'application/xml')]);
+
+                $formData = new FormDataPart($fields);
+
+                $response = $this->httpClient->request('POST', 'uploads', [
+                    'body' => $formData->bodyToIterable(),
+                    'headers' => $formData->getPreparedHeaders()->toArray(),
+                ]);
+
+                if (201 !== $statusCode = $response->getStatusCode()) {
+                    $this->logger->error(sprintf('Unable to upload translations for domain "%s" to phrase: "%s".', $domain, $response->getContent(false)));
+
+                    $this->throwProviderException($statusCode, $response, 'Unable to upload translations to phrase.');
+                }
+            }
+        }
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $keys = [[]];
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            foreach ($catalogue->getDomains() as $domain) {
+                $keys[] = array_keys($catalogue->all($domain));
+            }
+        }
+
+        $keys = array_unique(array_merge(...$keys));
+        $names = array_map(static fn ($v): ?string => preg_replace('/([\s:,])/', '\\\\\\\\$1', $v), $keys);
```

Comment:
don't you need to also replace backslashes?

Reply:
> for text queries on exact key names - spaces, commas, and colons need to be escaped with double backslashes
https://developers.phrase.com/api/#delete-/projects/-project_id-/keys

don't think so, at least not according to their documentation